### PR TITLE
Filter find by attributes and allow filter with multiple attributes

### PR DESCRIPTION
**Pull request by jay9645**
Improve attribute filtering to allow multiple attributes, and add filtering attributes to find command.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -16,28 +19,97 @@
...
+        this.attributes = attributes;
+    }
+
+    public boolean isAttributeSpecified() {
+        return !this.attributes.isEmpty();
+    }
+
+    private String getAttributesString() throws CommandException {
+        StringBuilder attributeName = new StringBuilder();
+        boolean isFirst = true;
```
Comment (wongkokian) [inline src/main/java/seedu/address/logic/commands/FindCommand.java]: Suggestion to name this isFirst to isFirstAttribute to make it more understandable.

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 2]
```diff
@@ -19,56 +20,75 @@
...
-        if (!this.isAttributeSpecified()) {
-            model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-            return new CommandResult(MESSAGE_SUCCESS);
-        } else {
-            model.updatePersonListByAttribute(this.attribute.get());
-            String attributeName = "";
-            switch (this.attribute.get()) {
+    private String getAttributesString() throws CommandException {
+        StringBuilder attributeName = new StringBuilder();
+        boolean isFirst = true;
```
Comment (wongkokian) [inline src/main/java/seedu/address/logic/commands/ListCommand.java]: Same here for the isFirst boolean as mentioned above.

**Review by wongkokian — APPROVED** [review]
Overall, well done on the enhancement. Good use of SLAP principle. There was attention to detail as well with the 'attribute' and 'attributes' differences. Just a suggestion to change the boolean in the code to make it slightly more readable, otherwise the flow is quite logical. LGTM 👍

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 1 above)
Comment (jay9645) [reply src/main/java/seedu/address/logic/commands/FindCommand.java]: Good point, done!

**Pull request merged**